pkg/util/log/format_crdb_v2.go

```diff
@@ -447,3 +454,147 @@ func (buf *buffer) maybeMultiLine(
 var startRedactionMarker = string(redact.StartMarker())
 var endRedactionMarker = string(redact.EndMarker())
 var markersStartWithMultiByteRune = startRedactionMarker[0] >= utf8.RuneSelf && endRedactionMarker[0] >= utf8.RuneSelf
+
+var entryREV2 = regexp.MustCompile(
+	`(?m)^` +
+		/* Severity                 */ `([` + severityChar + `])` +
+		/* Date and time            */ `(\d{6} \d{2}:\d{2}:\d{2}.\d{6}) ` +
+		/* Goroutine ID             */ `(?:(\d+) )` +
+		/* Go standard library flag */ `(\(gostd\) )?` +
+		/* Channel                  */ `(?:(\d+)@)?` +
+		/* File                     */ `([^:]+):` +
+		/* Line                     */ `(?:(\d+) )` +
+		/* Redactable flag          */ `((?:` + redactableIndicator + `)?) ` +
+		/* Context tags             */ `(?:\[((?:[^]]|\][^ ])+)\] )` +
+		/* Counter                  */ `((?:\d+)?) ` +
+		/* Continuation marker      */ `([ =!+|])` +
+		/* Message                  */ `(.*)$`,
+)
+
+// DecodeV2 decodes the next log entry into the provided protobuf message.
+func (d *EntryDecoder) DecodeV2(entry *logpb.Entry) error {
+	m := d.re.FindSubmatch(d.data)
+	if m == nil {
+		return io.EOF
+	}
```

Comment:
If this line is non-empty this indicates corruption of the log file rather than the end, right?

Reply:
Agreed - fixed.